Bio/motifs/__init__.py

```diff
@@ -237,10 +237,17 @@ def search(self, sequence):
 
     def reverse_complement(self):
         """Compute reverse complement of sequences."""
+        from Bio.Seq import Seq, MutableSeq
+        from Bio.SeqRecord import SeqRecord
+
         instances = Instances(alphabet=self.alphabet)
         instances.length = self.length
         for instance in self:
-            instance = instance.reverse_complement()
+            # TODO: remove inplace=False
+            if isinstance(instance, (Seq, MutableSeq)):
+                instance = instance.reverse_complement(inplace=False)
+            if isinstance(instance, (str, SeqRecord)):
+                instance = instance.reverse_complement()
```

Comment:
This looks wrong - what if the instance is none of the above? You won't attempt to take a reverse complement, nor raise an exception.

Reply:
Thanks. I agree that it's better to raise an exception here if it is none of the above. The instances were already checked when the object was created, but a user might have added more instances afterwards.